pkg/controller/hibernation/hibernation_controller.go

```diff
@@ -185,13 +199,22 @@ func (r *hibernationReconciler) Reconcile(ctx context.Context, request reconcile
 	}
 
 	hibernatingCondition := controllerutils.FindClusterDeploymentCondition(cd.Status.Conditions, hivev1.ClusterHibernatingCondition)
+	readyCondition := controllerutils.FindClusterDeploymentCondition(cd.Status.Conditions, hivev1.ClusterReadyCondition)
 
 	if supported, msg := r.hibernationSupported(cd); !supported {
 		// set hibernating condition to false for unsupported clouds
-		return r.setHibernatingCondition(cd, hivev1.UnsupportedHibernationReason, msg, corev1.ConditionFalse, cdLog)
+		changed := r.setHibernatingCondition(cd, hivev1.UnsupportedHibernationReason, msg, corev1.ConditionFalse, cdLog)
+		if changed {
+			err := r.updateClusterDeploymentStatus(cd, cdLog)
+			return reconcile.Result{}, err
+		}
```

Comment:
Due to only setting one condition now, there is a possibility that we don't set the powerstate immediately for these kind of situations - while I do believe it would be set eventually - in following reconciles, I'm not sure if that is good design. Any suggestions?